### PR TITLE
[refine](time) use TimeValue to refine repetitive logic and remove unused code.

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/functions/function_cast.h [hunk 1]
```diff
@@ -428,7 +429,6 @@ struct ConvertImpl {
                     for (size_t i = 0; i < size; ++i) {
                         (*vec_null_map_to)[i] = !TimeCast::try_parse_time(
                                 vec_from[i], vec_to[i], context->state()->timezone_obj());
-                        vec_to[i] *= (1000 * 1000);
```
Comment (HappenLee) [inline be/src/vec/functions/function_cast.h:431]: why delete the code？

Comment (Mryange) [reply be/src/vec/functions/function_cast.h:431]: TimeCast::try_parse_time handles this part.

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by BiteTheDDDDt — APPROVED** [review]

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**